javascripts/discourse/initializers/disco-toc-main.js

```diff
@@ -150,30 +161,67 @@ export default {
     }
   },
 
-  insertTOC(headings) {
+  createMainDiv() {
     const dToc = document.createElement("div");
     dToc.classList.add("d-toc-main");
-    dToc.innerHTML = `<div class="d-toc-icons">
-              <a href="#" class="scroll-to-bottom" title="${I18n.t(
-                themePrefix("post_bottom_tooltip")
-              )}">${iconHTML("downward")}</a>
-              <a href="#" class="d-toc-close">${iconHTML("times")}</a></div>`;
+    dToc.innerHTML = `<div class="d-toc-icons"><a href="#" class="d-toc-close">${iconHTML(
+      "times"
+    )}</a></div>`;
+    return dToc;
+  },
+
+  createScrollLink() {
+    const scrollLink = document.createElement("a");
+    scrollLink.href = "#";
+    scrollLink.className = "scroll-to-bottom";
+    scrollLink.title = I18n.t(themePrefix("post_bottom_tooltip"));
+    scrollLink.innerHTML = `${iconHTML("downward")} ${I18n.t(
+      themePrefix("jump_bottom")
+    )}`;
+    return scrollLink;
+  },
+
+  createToggleButton() {
+    const toggleButton = document.createElement("button");
+    toggleButton.className =
+      "d-toc-timeline-toggle btn btn-default btn-icon-text";
+    toggleButton.innerHTML =
+      iconHTML("timeline") + I18n.t(themePrefix("topic_timeline"));
+
+    return toggleButton;
+  },
```

Comment:
I'm not very familiar with the implementation of this theme but do you know why we're creating elements in Javascript instead of conditionally rendering them in Ember?